### ISSUE TITLE
Should a Pull Request to master be made for the domain-assertions-modularization branch ?

### ISSUE THREAD

**Issue by rhohimer** [ucoProject/UCO#149]
In Issue #147 a prototype branch was made to separate the domain assertion axioms from the ontology by placing them in a separate ontology and namespace.

This was done so that an application that uses a description logics reasoner could use the UCO without including that module.

The Question:
Should this branch be merged into master?

**Comment by rhohimer** [comment]
Of course those wishing to use the UCO ontology with those axioms can leave the domain-assertions ontology and namespace in place.

@ajnelson-nist  @sbarnum @HamptonJ  Thoughts?

**Comment by rhohimer** [comment]
If the V0.4.0 branch is more appropriate the pull request could be to it rather than master.

**Comment by ajnelson-nist** [comment]
Ah, I forgot to provide thoughts on this, sorry about that.
Thought 1:
We collectively still need to review/revise and sign off that document that defines the `develop` branch.
Thought 2:
This potentially has an impact on the ability to import just a piece of the UCO ontology.  If implemented as one .ttl file that includes all domain definitions, any need for even a single domain definition means a user needs to import all of UCO.  Splitting domain definitions into different files addresses that, but seems like a strange split (to me, anyway).
I do still support the objective of keeping domain definitions from hindering reasoners.

**Comment by rhohimer** [comment]
I definitely agree that splitting the domain assertion out this way is a bit strange. However, without doing so the UCO ontology was unusable by our reasoning application. It disturbs me to not be using the released branch of UCO. But we at DarkLight have no choice, we must not import the domain assertions.

**Comment by sbarnum** [comment]
Sorry for delayed response. 
I definitely agree with goal of supporting reasoners.
We just need to be very careful we don't break or make things very difficult for more mundane ( :-) ) use cases.
Splitting out the domain assertions like this may be an effective solution.
I have a primary and a secondary concern.
My primary concern is whether splitting them out this way even though semantically equivalent in any way prevents or impedes autogenerated documentation of the ontology which will be necessary. I am not in any way saying it does but I have not had a chance to test it and believe we need to do that before we can approve this approach.
My secondary concern is whether or not this is necessary. To my understanding, the reasoner issues you are having with the current ontology are related to a few cases where classes have multiple domains defined. There may be other general solutions to this issue such as using SHACL to define shapes for the classes. SHACL would likely provide benefit either way.

Regardless, I also concur with Alex's comment that if we do break out the domain assertions into separate TTL from the rest of the ontology they will need to be broken out into separate files according to namespace and form a pairing with the TTL files for the rest of ontology. This is necessary to support the granular use of the ontology.

**Comment by rhohimer** [comment]
In off-line discussions (out side this issue tracker) the decision was made to separate the domain assertions out into separate ontologies _per module_. For example, the action.ttl ontology file will have an action-da.ttl  ontology file which defines the domain assertions for the action ontology.

**Comment by rhohimer** [comment]
I will be uploading a new branch that separates the domain assertions out per module. The branch name will refer to this issue number (#149)
